### ISSUE TITLE
AppRail HTML markup is invalid and not accessible

### ISSUE THREAD

**Issue by is-sematell** [skeletonlabs/skeleton#2779]
### Current Behavior

The HTML code generated by AppRail produces invalid HTML and does not meet WCAG 2.0 parsing compliance for accessibility. E.g.
```html
<button class="app-rail-wrapper flex flex-col justify-center items-stretch w-full aspect-square undefined bg-primary-hover-token bg-primary-active-token" tabindex="0">
   <div class="h-0 w-0 overflow-hidden"><input type="radio" name="tile-1" value="0" tabindex="-1"></div>
   <div class="app-rail-interface text-center space-y-1">
      <div class="app-rail-lead "><i class="fa-solid fa-image text-2xl"></i></div>
      <div class="app-rail-label font-bold text-xs "><span>Tile 1</span></div>
   </div>
</button>
```

### Expected Behavior

AppRail should produce valid HTML and meet WCAG 2.0 parsing compliance.

### Steps To Reproduce

1. Use the AppRail component
2. Run the [Nu Html Checker](https://validator.w3.org/nu/) on the resulting code - [bookmarklets are also available](https://validator.w3.org/nu/about.html)
3. See errors generated

### More Information

The issue can be seen when validating the AppRail documentation page: https://validator.w3.org/nu/?showsource=yes&doc=https%3A%2F%2Fwww.skeleton.dev%2Fcomponents%2Fapp-rail

**Comment by endigo9740** [comment]
Just echoing the same answer from this other thread. This should address the issue reported:

- https://github.com/skeletonlabs/skeleton/issues/2780#issuecomment-2259189255
